raspberry/bzzz/sensors/pressure_sensor.py

```diff
@@ -1,210 +1,236 @@
-    
-    def update_altitude(self):
-        self._update_pressure()
-        self._calculate_altitude_from_pressure()
-        if self.log_file is not None:
-    
-            current_timestamp = datetime.datetime.now()
-            data_to_log = (self.pressure, self.altitude)
-            self.logger.record(current_timestamp, data_to_log)
-            time.sleep(0.5)
-   
+            return self.__data_processor.process(self.__values_cache[:, :], cursor=self.__cursor)[1]
+
+    def __pressure_temp_from_sensor(self):
+        """
+        Read raw pressure and temperature data from the BMP180 sensor.
+
+        This method communicates with the BMP180 sensor over the I2C bus to fetch the raw
+        temperature and pressure readings. It then uses the sensor's calibration data to
+        convert these raw readings into true temperature (in Celsius) and pressure (in Pa)
+        values based on the sensor's calibration coefficients.
+
+        The method handles the entire process of sending the correct commands to the sensor,
+        reading the raw data, and applying the necessary conversion formulas as outlined
+        in the BMP180 datasheet.
+
+        :return: A tuple containing the true pressure (in Pa) and temperature (in Celsius).
+        """
+        # Reading calibration data
+        data = self.__bus.read_i2c_block_data(self.__DEVICE_ADDRESS, 0xAA, 22)
+
+        # Converting calibration data
+        AC1 = data[0] * 256 + data[1]
+        if AC1 > 32767 :
+            AC1 -= 65535
+        AC2 = data[2] * 256 + data[3]
+        if AC2 > 32767 :
+            AC2 -= 65535
+        AC3 = data[4] * 256 + data[5]
+        if AC3 > 32767 :
+            AC3 -= 65535
+        AC4 = data[6] * 256 + data[7]
+        AC5 = data[8] * 256 + data[9]
+        AC6 = data[10] * 256 + data[11]
+        B1 = data[12] * 256 + data[13]
+        if B1 > 32767 :
+            B1 -= 65535
+        B2 = data[14] * 256 + data[15]
+        if B2 > 32767 :
+            B2 -= 65535
+        MB = data[16] * 256 + data[17]
+        if MB > 32767 :
+            MB -= 65535
+        MC = data[18] * 256 + data[19]
+        if MC > 32767 :
+            MC -= 65535
+        MD = data[20] * 256 + data[21]
+        if MD > 32767 :
+            MD -= 65535
+        # Reading raw temperature
+        self.__bus.write_byte_data(self.__DEVICE_ADDRESS, 0xF4, 0x2E)
+        time.sleep(0.005)
+        data = self.__bus.read_i2c_block_data(self.__DEVICE_ADDRESS, 0xF6, 2)
+        UT = data[0] * 256 + data[1]
+
+        # Reading raw pressure
+        self.__bus.write_byte_data(self.__DEVICE_ADDRESS, 0xF4, 0x74)
+        time.sleep(0.026)
+        data = self.__bus.read_i2c_block_data(self.__DEVICE_ADDRESS, 0xF6, 3)
+        UP = ((data[0] * 65536) + (data[1] * 256) + data[2]) / 128
+
+        # Calculating true temperature
+        X1 = (UT - AC6) * AC5 / 32768.0
+        X2 = (MC * 2048.0) / (X1 + MD)
+        B5 = X1 + X2
+        T = ((B5 + 8.0) / 16.0) / 10.0
```

Comment:
Change variable "T" to something more meaningful like "temperature" since we use it